demo/components/date-picker-demo.tsx

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { DatePicker } from '../../src';
+import {observable, action} from 'mobx';
+import {observer} from 'mobx-react';
+
+@observer
+export class DatePickerDemo extends React.Component<{}, {}> {
+    render () {
+        return <DatePicker placeholder="mm/dd/yyyy" />;
```

Comment:
DatePicker must be a controlled component, like the rest of the bunch. So it can't possibly work without some external state in the demo bound to the `value` and `onChange` props.